src/compile/output/mod.rs

```diff
@@ -516,10 +525,22 @@ impl AggregatedCompilerOutput {
     }
 
     /// Whether the output contains a compiler warning
-    pub fn has_warning(&self, ignored_error_codes: &[u64]) -> bool {
+    pub fn has_warning(&self, ignored_error_codes: &[u64], ignored_file_paths: &[PathBuf]) -> bool {
```

Comment:
I think we want a helper type for this, in case we want to fine-tune this in the future, for example "ignore these errors in these files"

I'd like something like

```rust
struct ErrorFilter<'a> {
  error_codes: Cow<'a, [u64]>,
  ignored_file_paths: Cow<'a, [PathBuf]>,
}
```

and a bunch of helper functions and From impls

so we can try to:

```rust
fn has_warning<'a>(&self, filter: impl Into<ErrorFilter<'a>)
```
perhaps, maybe not possible due to lifetime